compiler/rustc_infer/src/infer/opaque_types.rs

```diff
@@ -473,17 +473,6 @@ where
                 }
             }
 
-            ty::Alias(ty::Projection, proj) if self.tcx.is_impl_trait_in_trait(proj.def_id) => {
```

Comment:
In the new lowering scheme, we should never need to consider the variances of a RPITIT GAT.